src/util/system.h

```diff
@@ -40,6 +40,11 @@ int64_t GetStartupTime();
 
 extern const char * const BITCOIN_CONF_FILENAME;
 extern const char * const BITCOIN_PID_FILENAME;
+#ifdef ENABLE_WALLET
```

Comment:
Please don't add `#ifdef ENABLE_WALLET` in the utility code! We've only recently painstakingly removed `#ifdef ENABLE_WALLET` stuff from the `server` library, for example.

Also I think it probably should be switched based on the run-time `-disablewallet` flag as well?

Or maybe: always enable it, don't make it depend on the wallet at all? What would be the drawback to that?

Reply:
@laanwj Performance slowdown. For a node not running a wallet having mitigation for spectre seems rather pointless, but I could be wrong on that.

Reply:
Should I move that to somewhere else?

Reply:
What *is* the performance slowdown in practice?
I think this needs benchmarking, if it's neglible over say, a full sync, I don't think it makes sense to do this kind of switching at all.

Reply:
Perhaps this should be a ParameterInteraction thing using SoftSetArg in the wallet logic?